core/src/main/java/org/kiwix/kiwixmobile/core/page/history/HistoryFragment.kt

```diff
@@ -22,8 +27,9 @@ class HistoryActivity : PageActivity() {
   override val switchString: String by lazy { getString(R.string.history_from_current_book) }
   override val title: String by lazy { getString(R.string.history) }
   override val switchIsChecked: Boolean by lazy { sharedPreferenceUtil.showHistoryAllBooks }
-  override fun injection(coreComponent: CoreComponent) {
-    activityComponent.inject(this)
+
+  override fun inject(baseActivity: BaseActivity) {
+    requireActivity().coreActivityComponent.inject(this)
```

Comment:
cachedComponent in the inject calls too

Reply:
Is it ok if `Custom` & `KiwixActivityComponent` implements the `CoreActivityComponent`, or should I create a new `CoreActivityComponentInterface` so that `CoreMainActivity` can use an abstract `cachedComponent`?

Reply:
I would need to look at the code a lot, depends on what dagger is happy with because it is rarely happy about anything